src/tools/thmgen/thmgen.cs

```diff
@@ -0,0 +1,165 @@
+    /// </summary>
+    public sealed class ThmGen
+    {
+        private ThmGenCommandLine commandLine;
+        private XNamespace NameSpace = "http://wixtoolset.org/schemas/v4/thmutil";
+
+        /// <summary>
+        /// The main entry point for ThmGen.
+        /// </summary>
+        /// <param name="args">Command-line arguments for the application.</param>
+        /// <returns>Returns the application error code.</returns>
+        [MTAThread]
+        public static int Main(string[] args)
+        {
+            AppCommon.PrepareConsoleForLocalization();
+            Messaging.Instance.InitializeAppName("THMG", "thmgen.exe").Display += AppCommon.ConsoleDisplayMessage;
+
+            ThmGen thmgen = new ThmGen();
+            return thmgen.Run(args);
+        }
+
+        /// <summary>
+        /// Main running method for the application.
+        /// </summary>
+        /// <param name="args">Command-line arguments to the application.</param>
+        /// <returns>Returns the application error code.</returns>
+        private int Run(string[] args)
+        {
+            try
+            {
+                this.commandLine = new ThmGenCommandLine();
+                string[] unprocessed = commandLine.Parse(args);
+
+                if (!Messaging.Instance.EncounteredError)
+                {
+                    if (this.commandLine.ShowLogo)
+                    {
+                        AppCommon.DisplayToolHeader();
+                    }
+
+                    if (this.commandLine.ShowHelp)
+                    {
+                        Console.WriteLine(ThmGenStrings.HelpMessage);
+                        AppCommon.DisplayToolFooter();
+                    }
+                    else
+                    {
+                        foreach (string arg in unprocessed)
+                        {
+                            Messaging.Instance.OnMessage(WixWarnings.UnsupportedCommandLineArgument(arg));
+                        }
+
+                        this.GenerateThemeHelpers();
+                    }
+                }
+            }
+            catch (WixException we)
+            {
+                Messaging.Instance.OnMessage(we.Error);
+            }
+            catch (Exception e)
+            {
+                Messaging.Instance.OnMessage(WixErrors.UnexpectedException(e.Message, e.GetType().ToString(), e.StackTrace));
+                if (e is NullReferenceException || e is SEHException)
+                {
+                    throw;
+                }
+            }
+
+            return Messaging.Instance.LastErrorNumber;
+        }
+
+        private void GenerateThemeHelpers()
+        {
+            var outputHeaderFile = Path.ChangeExtension(this.commandLine.OutputFile ?? "theme.h", ".h");
+            var outputModuleFile = Path.ChangeExtension(outputHeaderFile, ".cpp");
+            var prefix = this.commandLine.Prefix ?? String.Empty;
+            var pch = this.commandLine.PchName ?? "stdafx.h";
```

Comment:
Might consider just making this the "header to get dutil stuff included". Precompiled header is the most common case...